### PR TITLE
Improve default config file

### PR DESCRIPTION
**Pull request by curquiza**
Following https://github.com/meilisearch/meilisearch/pull/2745#pullrequestreview-1115361274

@gmourier @maryamsulemani97, here are the changes I applied
- [After discussing with the cloud team](https://meilisearch.slack.com/archives/C03T1T47TUG/p1666014688582379) (internal link only), I let commented and uncommented the variables depending on their role.
- The only boolean variable I let be commented is the `no_analytics` one, and I wrote the non-default value (i.e. `true`)-> it will be easier for the users to disable analytics by only uncommenting the line. Plus, `no_analytics = false` is a double negative and not really easy to understand.
- I removed the `INDEX` section that was not really related to indexes
- I keep only 4 sections: SSL, Dumps, Snapshots, and an untitled section with all the uncategorized variables. I gathered all the uncategorized variable at the top of the file, so in the "untitled" section.
- I copied/pasted the really basic [documentation description](https://docs.meilisearch.com/learn/configuration/instance_options.html) of the variables
- I added every doc linked to each variable. 

Let me know if you agree or not with my choices!
Thanks for your review 🙏

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -1,129 +1,135 @@
...
-# This environment variable must be set to `production` if you are running in production.
-# More logs wiil be displayed if the server is running in development mode. Setting the master
-# key is optional; hence no security on the updates routes. This
-# is useful to debug when integrating the engine with another service.
+env = "development"
+# Configures the instance's environment. Value must be either `production` or `development`.
+# https://docs.meilisearch.com/learn/configuration/instance_options.html#environment
 
-http_addr = "127.0.0.1:7700"
+http_addr = "localhost:7700"
```
Comment (curquiza) [inline config.toml:12]: Changed following https://github.com/meilisearch/meilisearch/issues/2782

File: config.toml [hunk 2]
```diff
@@ -1,129 +1,135 @@
...
 # The address on which the HTTP server will listen.
 
-# master_key = "MASTER_KEY"
+# master_key = "YOUR_MASTER_KEY_VALUE"
```
[the remark this answers is not in the document]
Comment (curquiza) [reply config.toml:15]: line 19 is `no_analytics` -> I wrote my justification in the PR description, maybe it's not clear 🫣 let me know!
line 42 is `max_indexing_memory` -> the value I wrote is not the default value. The default value of `max_indexing_memory` is 2/3 of the RAM. That's why I cannot uncomment it
line 46 is `max_indexing_threads` -> same, the default value is not 4 threads but half of the core, so I cannot uncomment it

[the remark this answers is not in the document]
Comment (curquiza) [reply config.toml:15]: For the other commented one -> if I uncomment them, I impose something not mandatory when launching meilisearch, like importing a snapshot or a dump

[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply config.toml:15]: Okay, it makes sense now. Thanks 🙏

**Review by maryamsulemani97 — APPROVED** [review]
LGTM 🏃🏻‍♀️

**Comment by curquiza** [comment]
bors merge

**Pull request merged**